File: app/src/main/java/com/otraempresa/android/example/AbstractRequestTokenAsyncTask.java
package com.otraempresa.android.example;

import android.content.Context;
import android.os.AsyncTask;
import android.widget.Toast;

import com.kushkipagos.android.Card;
import com.kushkipagos.android.Kushki;
import com.kushkipagos.android.KushkiEnvironment;
import com.kushkipagos.android.KushkiException;
import com.kushkipagos.android.Transaction;

abstract class AbstractRequestTokenAsyncTask extends AsyncTask<Card, Void, Transaction> {

    protected final Kushki kushki;
    protected final Kushki kushkiCardAsync;
    private final Context context;

    AbstractRequestTokenAsyncTask(Context context) {
        this.context = context;
        kushki = new Kushki("10000001656015280078454110039965", "USD", KushkiEnvironment.TESTING);
        kushkiCardAsync = new Kushki("20000000103098876000", "CLP", KushkiEnvironment.QA);
    }

    @Override
    protected Transaction doInBackground(Card... args) {
        try {
            return requestToken(args[0]);
        } catch (KushkiException kushkiException) {
            throw new RuntimeException(kushkiException);
        }
    }

    @Override
    protected void onPostExecute(Transaction transaction) {
        if (transaction.isSuccessful()) {
            showToast(transaction.getToken());
        } else {
            showToast("ERROR: " + transaction.getCode() + " " + transaction.getMessage());
        }
    }

    protected abstract Transaction requestToken(Card card) throws KushkiException;

    private void showToast(String text) {
        Toast toast = Toast.makeText(context, text, Toast.LENGTH_SHORT);
        toast.show();
    }
}

File: app/src/main/java/com/kushkipagos/kushkidemo/contracts/AbstractRequestTokenAsyncTask.kt
package com.kushkipagos.kushkidemo.contracts

import android.content.Context
import android.widget.Toast
import com.kushkipagos.exceptions.KushkiException
import com.kushkipagos.kushki.Kushki
import com.kushkipagos.kushki.KushkiEnvironment
import com.kushkipagos.kushkidemo.helpers.CoroutineTask
import com.kushkipagos.models.Card
import com.kushkipagos.models.Transaction

abstract class AbstractRequestTokenAsyncTask : CoroutineTask<Card, Void, Transaction> {
    protected var _kushki: Kushki
    private var _context: Context

    constructor(context: Context){
        _context = context
        _kushki = Kushki("1195b05f49b44aae83fafc9c8de01b88", "COP", KushkiEnvironment.QA)
    }

    @Throws(KushkiException::class)
    abstract fun requestToken(card: Card) : Transaction

    override fun doInBackground(vararg params: Card?): Transaction {
        try {
            return requestToken(params[0]!!);
        } catch (kushkiException: KushkiException) {
            throw RuntimeException(kushkiException);
        }
    }

    override fun onPostExecute(param: Transaction?) {
        val transaction = param!!
        if (transaction.isSuccessful) {
            showToast("TOKEN: ${transaction.token} SECURE_ID: ${transaction.secureId}")
        } else {
            showToast("ERROR: Code: ${transaction.code} ${transaction.message}")
        }
    }

    fun showToast(text: String){
        val toast = Toast.makeText(_context, text, Toast.LENGTH_LONG)
        toast.show()
    }
}